server.js

```diff
@@ -1658,22 +1661,23 @@ cache(function(data, match, sendBadge, request) {
 }));
 
 // npm node version integration.
-camp.route(/^\/node\/v\/(?:@([^/]+))?\/?([^/]*)\/?([^/]*)\.(svg|png|gif|jpg|json)$/,
+camp.route(/^\/node(?:\/(http(?:s)?)\/([^/]+))?\/v\/(?:@([^/]+))?\/?([^/]*)\/?([^/]*)\.(svg|png|gif|jpg|json)$/,
 cache(function(data, match, sendBadge, request) {
-  // e.g. @stdlib, stdlib, next, svg
-  const [, scope, packageName, tag, format] = match;
+  // e.g. https, registry.npmjs.org, @stdlib, stdlib, next, svg
+  const [, protocol, host, scope, packageName, tag, format] = match;
+  const registryUrl = getNpmRegistryUrl(protocol, host);
   const registryTag = tag || 'latest';
   let apiUrl;
   if (scope === undefined) {
       // e.g. https://registry.npmjs.org/express/latest
       // Use this endpoint as an optimization. It covers the vast majority of
       // these badges, and the response is smaller.
-      apiUrl = `https://registry.npmjs.org/${packageName}/${registryTag}`;
+      apiUrl = `${registryUrl}/${packageName}/${registryTag}`;
   } else {
     // e.g. https://registry.npmjs.org/@cedx%2Fgulp-david
     // because https://registry.npmjs.org/@cedx%2Fgulp-david/latest does not work
     const path = encodeURIComponent(`${scope}/${packageName}`);
-    apiUrl = `https://registry.npmjs.org/@${path}`;
+    apiUrl = `${registryUrl}/@${path}`;
```

Comment:
I think I wrote tests for the current Node and NPM badges. Could you add tests of the new routes?

Reply:
sure

Reply:
done